spec/stamp_spec.cr

```diff
@@ -0,0 +1,120 @@
+require "./spec_helper"
+
+describe Hashcash::Stamp do
+  it ".new" do
```

Comment:
When describing a set of functionality, use `describe` (you can nest describe blocks) then place individual tests (`it` blocks) inside this:
```crystal
describe Hashcash::Stamp do
  describe ".initialize" do
    it "accepts just a resource arg" do
      new_stamp = Hashcash::Stamp.new("gab@palce.technology")
      #...
    end
    
    it "accepts a full set of stamp parameters" do
      custom_stamp = Hashcash::Stamp.new("hi@hello.com", "2", 16, Time.utc, "goodbye")
      #...
    end
  end
end
```
This will provide a nice granular output so that others (or future you) working on this can quickly identify any issues.